clippy_lints/src/booleans.rs

```diff
@@ -346,11 +346,17 @@ fn simplify_not(cx: &LateContext<'_>, expr: &Expr<'_>) -> Option<String> {
                 _ => None,
             }
             .and_then(|op| {
-                Some(format!(
-                    "{}{op}{}",
-                    snippet_opt(cx, lhs.span)?,
-                    snippet_opt(cx, rhs.span)?
-                ))
+                let lhs_snippet = snippet_opt(cx, lhs.span)?;
+                let rhs_snippet = snippet_opt(cx, rhs.span)?;
+
+                match &lhs.kind {
+                    ExprKind::Cast(..) if binop.node == BinOpKind::Ge => {
+                        // e.g. `(a as u64) < b`. Without the parens the `<` is
+                        // interpreted as a start of generic arguments for `u64`
+                        Some(format!("({lhs_snippet}){op}{rhs_snippet}"))
```

Comment:
Just run a quick test on your code with some of my theories, and confirmed that you might need to check if the parens exists before adding one.

right now it's suggesting a double parenthesis if there already one like:

```rust
if !((a as u64) >= b) {}
```

the fixed code becomes:

```rust
if ((a as u64)) < b {}
```